File: test/fixtures/ruby/get_referer.rb
require 'net/http'

uri = URI('http://localhost:28139')
req = Net::HTTP::Get.new(uri)
req['X-Requested-With'] = 'XMLHttpRequest'
req['User-Agent'] = 'SimCity'
req['Referer'] = 'https://website.com'

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/java/get_referer.java
import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

HttpClient client = HttpClient.newHttpClient();

HttpRequest request = HttpRequest.newBuilder()
    .uri(URI.create("http://localhost:28139"))
    .GET()
    .setHeader("X-Requested-With", "XMLHttpRequest")
    .setHeader("User-Agent", "SimCity")
    .setHeader("Referer", "https://website.com")
    .build();

HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
